packages/cli/src/info/cdk_info_provider.ts

```diff
@@ -0,0 +1,67 @@
+import { execa as _execa } from 'execa';
+import { format } from '@aws-amplify/cli-core';
+
+/**
+ * Provides information about CDK.
+ */
+export class CdkInfoProvider {
+  /**
+   * execa to run the cdk command and testing.
+   */
+  constructor(private readonly execa = _execa) {}
+
+  /**
+   * Format CDK information.
+   * @returns The cdk doctor output.
+   */
+  async getCdkInfo(): Promise<string> {
+    const cdkDoctorArgs: string[] = ['cdk', 'doctor', '--', ' --no-color'];
+
+    const output = await this.execa('npx', cdkDoctorArgs, {
+      all: true,
+    });
+
+    return this.formatCdkInfo(output.all ?? output.stderr);
+  }
+
+  /**
+   * Format CDK information.
+   * @param info - The CDK information to format.
+   * @returns The formatted CDK information.
+   */
+  private formatCdkInfo(info: string): string {
+    const sensitiveKeys = [
+      'AWS_ACCESS_KEY_ID',
+      'AWS_SECRET_ACCESS_KEY',
+      'AWS_SESSION_TOKEN',
+    ];
+
+    const lines = info.split(/\r\n|\r|\n/);
+    const formattedLines = lines.map((line) => {
```

Comment:
we should start with filtering out sensitive keys before anything else
```suggestion
    const formattedLines = lines
      .filter(line => !sensitiveKeys.some(key => line.includes(key)))
      .map(line => {
        // rest of stuff
      })
```

Reply:
<img width="322" alt="image" src="https://github.com/aws-amplify/amplify-backend/assets/87995712/9c61ecc0-eb84-47d7-924b-35129d3cdedd">

by default the cdk command already redact the info on sensitive variables. filtering the variables is to ensure we have a cleaner output.